packages/core/src/lib/Utils.ts

```diff
@@ -1325,6 +1325,8 @@ async function closePuppeteer(
   if (config.isLocalPuppeteer && !options.warmup) {
     await Promise.all(pages.map(page => page.close()));
     await browser.disconnect();
+  } else if (config.skipBrowserCloseWait === true) {
+    browser.close();
```

Comment:
In this case, skipping waiting for the promising to resolve may allow memlab to proceed, but after memory leak detection is done, the JS runtime would still not able to terminate since it would wait for all promises to resolve. Do you force terminate the node.js runtime somehow in the end?

Reply:
Thanks for the question!
We use Memlab inside AWS Lambdas where we do not have to wait for the runtime to finish. It keeps running and in the meantime if the browser finally closes then good, if not then AWS Lambda will terminate the runtime anyway. Ideally, of course, would be the browser.close() to not hang and just close the browser properly, but it's not so straightforward why the closing hangs, it would require deeper investigation.

In some cases iterating through all the opened pages and closing them before closing the browser worked, but in case of Memlab this does not seem to work correctly as the browser instance seems to return no opened pages.
https://github.com/Sparticuz/chromium/issues/85#issuecomment-1549894734  

Please let me know if I explained this fine or would you want further clarification. Happy to discuss further if needed